pyxem/signals/insitu_diffraction2d.py

```diff
@@ -0,0 +1,339 @@
+                                             depth=overlapped_depth,
+                                             boundary={a: 'none' for a in range(5)})
+
+        # Clone original overlap dask array to work around memory release issue in map_overlap
+        data_clones = da.concatenate(
+            [clone(b, omit=data_overlapped) for b in data_overlapped.blocks]
+        )
+
+        mapped = data_clones.map_blocks(_register_drift_5d,
+                                        shifts1=xdrift_dask,
+                                        shifts2=ydrift_dask,
+                                        dtype='float32')
+
+        registered_data = InSituDiffraction2D(
+            da.overlap.trim_internal(mapped, overlapped_depth)
+        ).as_lazy()
+
+        # Set axes info for registered signal
+        for nav_axis_old, nav_axis_new in zip(
+                s_.axes_manager.navigation_axes, registered_data.axes_manager.navigation_axes
+        ):
+            pst._copy_axes_object_metadata(nav_axis_old, nav_axis_new)
+        for sig_axis_old, sig_axis_new in zip(
+                s_.axes_manager.signal_axes, registered_data.axes_manager.signal_axes
+        ):
+            pst._copy_axes_object_metadata(sig_axis_old, sig_axis_new)
+
+        if not lazy_result:
+            registered_data.compute()
+
+        return registered_data
+
+    def correct_real_space_drift_fast(self, shifts=None, time_axis=2, lazy_result=True):
+        """
+        Perform real space drift registration on the dataset with fast performance
+        over spatial axes. If signal is lazy, spatial axes must not be chunked
+
+        Parameters
+        ----------
+        shifts: Signal1D
+            shift vectors to register, must be in the shape of <N_time | 2>.
+            If None, shift vectors will be calculated automatically
+        time_axis: int
+            Index of time axis. Default is 2
+        lazy_result: bool, default True
+            Whether to return lazy result. Only relevant if signal is not lazy
+
+        Returns
+        ---------
+        registered_data: InSituDiffraction2D
+            Real space drift corrected version of the original dataset
+        """
+        if self._lazy:
+            nav_axes = [0, 1, 2]
+            nav_axes.remove(2 - time_axis)
+            chunkings = self.get_chunk_size()
+            if len(chunkings[nav_axes[0]]) != 1 or len(chunkings[nav_axes[1]]) != 1:
+                raise Exception("Spatial axes are chunked. Please rechunk signal or use \'correct_real_space_drift\' "
+                                "instead")
+
+        if shifts is None:
+            shifts = self.get_drift_vectors(time_axis=time_axis)
+
+        if time_axis != 2:
+            s_ = self.roll_time_axis(time_axis=time_axis)
+        else:
+            s_ = self
+        s_transposed = s_.transpose(signal_axes=(0, 1))
+
+        xdrift = shifts.data[:, 0]
+        ydrift = shifts.data[:, 1]
+        xs = Signal1D(
```

Comment:
So I think that this is overkill. 

The `map` function will copy arrays to every single function so you should just be able to pass:

```
s_transposed.map(_register_drift_2d,
                                            shift1=xdrift,
                                            shift2=ydrift,
                                            inplace=False
                                            )
```
Without repeating the data.  The added benefit is that it will use less memory duplicating the shift array

Reply:
The `xdrift`/'ydrift' parameters are different for different time coordinates, which is a navigation axes. Therefore I need to pass it as signal that has the same navigation dimension as `s_transposed`.